subiquity/controllers/installprogress.py

```diff
@@ -291,18 +294,20 @@ def _bg_install_openssh_server(self):
 
     def _bg_cleanup_apt(self):
         if self.opts.dry_run:
-            cmd = [
-                "sleep", str(2/self.scale_factor),
-                ]
+            cmds = [["sleep", str(1/self.scale_factor)]]
         else:
-            os.unlink(
-                os.path.join(
-                    self.base_model.target, "etc/apt/sources.list.d/iso.list"))
-            cmd = [
-                sys.executable, "-m", "curtin", "in-target", "-t", "/target",
-                "--", "apt-get", "update",
+            cmds = [
+                ["umount", self.tpath('etc/apt')],
                 ]
-        self._bg_run_command_logged(cmd)
+            if self.base_model.network.has_network:
+                cmds.append([
+                    sys.executable, "-m", "curtin", "in-target", "-t",
+                    "/target", "--", "apt-get", "update",
```

Comment:
The shell program also calls apt-get update, are both needed?

Reply:
See above: this update is run after the apt configuration has been reverted to the configuration we want for the installed system so yes, it's needed.

Reply:
I see, we're refreshing the cache since the repos have changed.  That makes sense.